src/main/java/duke/command/TodoCommand.java

```diff
@@ -0,0 +1,62 @@
+package duke.command;
+
+import duke.exception.NoActionException;
+import duke.exception.SaveFileException;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.util.Reply;
+import duke.util.Storage;
+import duke.util.TaskList;
+
+/**
+ * A command class encapsulating the logic that occurs when the user issues a 'todo' command.
+ */
+public class TodoCommand extends Command {
+    private String action;
+
+    /**
+     * Constructor of the TodoCommand
+     *
+     * @param action String input of the tasks
+     */
+    public TodoCommand(String action) {
+        super(false);
+        this.action = action;
+    }
+
+    /**
+     * Creates and adds a Todo to the tasks.
+     *
+     * @param tasks List of existing tasks
+     * @param storage Storage class handling the persistence of the tasks
+     * @throws NoActionException if no action was provided
+     * @throws SaveFileException if there is a save file related error
+     */
+    @Override
+    public CommandResult execute(TaskList tasks, Storage storage)
+            throws NoActionException, SaveFileException {
```

Comment:
Can throws NoActionException be on the same line as method declaration? It might be clearer.